subsys/bluetooth/host/adv.c

```diff
@@ -444,15 +444,45 @@ static int hci_set_adv_ext_complete(struct bt_le_ext_adv *adv, uint16_t hci_op,
 	return bt_hci_cmd_send_sync(hci_op, buf, NULL);
 }
 
-static int hci_set_adv_ext_fragmented(struct bt_le_ext_adv *adv, uint16_t hci_op,
+static int hci_set_adv_ext_fragmented(const struct bt_le_ext_adv *adv, uint16_t hci_op,
```

Comment:
Smart move to use this for periodic advertising data too, but I do wonder if the name is still fitting. Doesn't necessarily need a change, but something to consider.

Reply:
No matter what the resolution is here, I consider the current implementation unacceptable as it is. Please take out the common parts into a common function that can be then used by the per-command functions like we do everywhere else, including hci.c.